### PR TITLE
Allow option to persist deleted values on delete

### PR DESCRIPTION
**Pull request by kevinquinnyo**
Reference issue: https://github.com/lorenzo/audit-stash/issues/19

We should probably add tests for the new Behavior configuration option `logDeletedValues`.

### CODE REVIEW

File: src/Event/AuditDeleteEvent.php [hunk 1]
```diff
@@ -23,13 +16,17 @@ class AuditDeleteEvent implements EventInterface
...
-        $this->source = $source;
+        if (is_array($parentSource) === true) {
+            $parentSource = null;
+            $original = $parentSource;
+        }
+
         $this->parentSource = $parentSource;
-        $this->timestamp = (new DateTime())->format(DateTime::ATOM);
+
+        //debug(compact('transactionId', 'id', 'source', 'parentSource', 'original')); die();
```
Comment (kevinquinnyo) [inline src/Event/AuditDeleteEvent.php:28]: delete the debug line

Comment (lorenzo) [reply src/Event/AuditDeleteEvent.php:28]: Can you please delete it?

File: src/Event/AuditDeleteEvent.php [hunk 2]
```diff
@@ -23,13 +16,17 @@ class AuditDeleteEvent implements EventInterface
      * @param string $source The name of the source (table) where the record was deleted
      * @param string $parentSource The name of the source (table) that triggered this change
      */
-    public function __construct($transactionId, $id, $source, $parentSource = null)
+    public function __construct($transactionId, $id, $source, $parentSource = null, $original = null)
     {
-        $this->transactionId = $transactionId;
-        $this->id = $id;
-        $this->source = $source;
+        if (is_array($parentSource) === true) {
```
Comment (lorenzo) [inline src/Event/AuditDeleteEvent.php:21]: I would rather not have this check at all and avoid messing up with the arguments.

**Comment by ishanvyas22** [comment]
Any update in this?

**Comment by lorenzo** [comment]
@ishanvyas22 no, but feel free to take over this code and send a new pull request

**Comment by ishanvyas22** [comment]
@lorenzo I've created new [PR](https://github.com/lorenzo/audit-stash/pull/50), please have a look.

**Pull request closed without merging**